torchvision/prototype/transforms/__init__.py

```diff
@@ -12,3 +12,14 @@
 from ._misc import Identity, Normalize, ToDtype, Lambda
 from ._presets import CocoEval, ImageNetEval, VocEval, Kinect400Eval, RaftEval
 from ._type_conversion import DecodeImage, LabelToOneHot
+
+# What are the migration plans for Classes without new API equivalents? There are two categories:
+# 1. Deprecated methods which have equivalents on the new API (_legacy.py?):
+# - Grayscale, RandomGrayscale: use ConvertImageColorSpace
+# 2. Those without equivalents on the new API:
+# - Pad, RandomCrop, RandomHorizontalFlip, RandomVerticalFlip, RandomPerspective, FiveCrop, TenCrop, ColorJitter,
+# RandomRotation, RandomAffine, GaussianBlur, RandomInvert, RandomPosterize, RandomSolarize, RandomAdjustSharpness,
+# RandomAutocontrast, RandomEqualize, LinearTransformation (must be added)
```

Comment:
:+1: For everything but `FiveCrop` and `TenCrop`. They are outliers in the sense that they get one input and produce multiple outputs. This kind of structure would be suited a lot better to a datapipe rather than a transform. 

This is also reinforced by the fact that these transforms can only be put at the end of any pipeline with the old API given that the output type is not compatible with all the other transforms. Even then, using such a transform would most likely need a custom collate function.

I couldn't find any usage in our reference scripts. Does someone know how exactly they are used so I can propose something more concrete here?

Reply:
The problem is if we want to roll out the API with BC support, we must to include implementations for all previous methods + offer alternatives for those that would be placed in deprecated state. Thought I understand the challenge, excluding these methods is not OK and we should flag if others like these exist.

The specific transforms are used to produce multiple crops for inference in classification. It is a common trick (especially in the early days of ImageNet competition) to do boost the accuracy of the model by averaging the responses of multiple crops. You can see an example of how this is used [here](https://github.com/pytorch/vision/blob/main/torchvision/transforms/transforms.py#L981-L989).